pyomo/contrib/solver/solvers/highs.py

```diff
@@ -96,6 +96,121 @@ def update(self):
         self.highs.changeColCost(col_ndx, value(self.expr))
 
 
+class _MutableQuadraticCoefficient:
+    def __init__(self, expr, row_idx, col_idx):
+        self.expr = expr
+        self.row_idx = row_idx
+        self.col_idx = col_idx
+
+
+class _MutableObjective:
+    def __init__(self, highs, constant, linear_coefs, quadratic_coefs):
+        self.highs = highs
+        self.constant = constant
+        self.linear_coefs = linear_coefs
+        self.quadratic_coefs = quadratic_coefs
+        self.last_quadratic_coef_values = [value(i.expr) for i in self.quadratic_coefs]
+        # Store the quadratic coefficients in dictionary format
+        self.quad_coef_dict = {}
+        self._initialize_quad_coef_dict()
+        # Flag to force first update of quadratic coefficients
+        self._first_update = True
+
+    def _initialize_quad_coef_dict(self):
+        for coef in self.quadratic_coefs:
+            v1_ndx = coef.row_idx
+            v2_ndx = coef.col_idx
+            # Ensure we're storing the lower triangular part
+            row = max(v1_ndx, v2_ndx)
+            col = min(v1_ndx, v2_ndx)
+
+            coef_val = value(coef.expr)
+            # Adjust for diagonal elements
+            if v1_ndx == v2_ndx:
+                coef_val *= 2.0
+
+            self.quad_coef_dict[(row, col)] = coef_val
+
+    def update(self):
+        """
+        Update the quadratic objective expression.
+        """
+        needs_quadratic_update = self._first_update
+
+        self.constant.update()
+        for coef in self.linear_coefs:
+            coef.update()
+
+        for ndx, coef in enumerate(self.quadratic_coefs):
+            current_val = value(coef.expr)
+            if current_val != self.last_quadratic_coef_values[ndx]:
+                needs_quadratic_update = True
+
+                v1_ndx = coef.row_idx
+                v2_ndx = coef.col_idx
+                row = max(v1_ndx, v2_ndx)
+                col = min(v1_ndx, v2_ndx)
+
+                # Adjust the diagonal to match Highs' expected format
+                if v1_ndx == v2_ndx:
+                    current_val *= 2.0
+
+                self.quad_coef_dict[(row, col)] = current_val
+
+                self.last_quadratic_coef_values[ndx] = current_val
+
+        # If anything changed, rebuild and pass the Hessian
+        if needs_quadratic_update:
+            self._build_and_pass_hessian()
+            self._first_update = False
+
+    def _build_and_pass_hessian(self):
+        """Build and pass the Hessian to HiGHS in CSC format"""
+        if not self.quad_coef_dict:
+            return
+
+        dim = self.highs.getNumCol()
+
+        # Build CSC format for the lower triangular part
+        q_value = []
```

Comment:
Does `q_*` stand for `quadratic_*`?

Reply:
The hessian is often referred to as a matrix Q (in many books I see a quadratic objective defined as 1/2 x.T @ Q @ x). I am happy to change to whatever you prefer

Reply:
I think I'd prefer `hessian_value` or `hess_value` then - we do use that pretty widely within Pyomo.